app/anti_virus_check.py

```diff
@@ -0,0 +1,139 @@
+import collections
+import json
+import os
+import time
+
+import requests
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError, BadMessageError
+
+from app import create_and_wrap_logger
+from app import settings
+
+logger = create_and_wrap_logger(__name__)
+
+AVResult = collections.namedtuple('AVResult', 'safe ready scan_results')
+
+
+class AntiVirusCheck:
+    def __init__(self, tx_id):
+        self.bound_logger = logger.bind(tx_id=tx_id)
+
+    def send_for_av_scan(self, payload):
+        self.bound_logger.debug("A/V scanned enabled {}".format(settings.ANTI_VIRUS_ENABLED))
+        self.bound_logger.info("Sending for AV check")
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        ready = False
+        attempts = 0
+        while not ready:
```

Comment:
- Is the ready boolean needed? Once it's set to `True` the code either raises a `QuarantinableError` or returns `True`.
- I think attempts should be incremented as soon as the loop is entered as that's when the attempt is started.

```Python
attempts = 0
while True:
    attempts += 1
    results = self._get_anti_virus_result(data_id)
    if not results.ready:
        if attempts >= settings.ANTI_VIRUS_MAX_ATTEMPTS:
            raise RetryableError()
        time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
    elif not results.safe:
	...
```